compiler/next/include/chpl/AST/Location.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_AST_LOCATION_H
+#define CHPL_AST_LOCATION_H
+
+#include "chpl/AST/UniqueString.h"
+
+namespace chpl {
+namespace ast {
+
+/**
+  This class represents a source location.
+ */
+class Location final {
+ private:
+  UniqueString path_;
+  int firstLine_;
+  int firstColumn_;
+  int lastLine_;
+  int lastColumn_;
+ public:
```

Comment:
In C++11 and later, you can specify member initializers like `int lastColumn_ = -1;` and they will apply if there is no initializer in the constructor. This may make the code more brief or easier to understand, if the default values of the members are always the same for every constructor. That way you don't have to repeat the default value over and over for every constructor, but can just leave it out of the constructor's initializer list, and the member's own initializer will be used.